### PR TITLE
[MIG]migration with commit history for product_code_unique module from v11 to V12

### PR DESCRIPTION
**Pull request by bizzappdev**
migration from v11 to 12
#389

### CODE REVIEW

File: product_code_unique/__init__.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Copyright (C) 2018 - TODAY, Open Source Integrators
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from . import models
+
+
+def pre_init_product_code(cr):
```
Comment (nikul-serpentcs) [inline product_code_unique/__init__.py]: @bizzappdev Here use ` hook file` like https://github.com/OCA/maintainer-tools/blob/master/template/module/hooks.py

File: product_code_unique/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+{
+    "name": "Unique Product Internal Reference",
+    "summary": "Set Product Internal Reference as Unique",
+    "version": "12.0.1.0.0",
+    "license": "AGPL-3",
+    "author": "Open Source Integrators, Odoo Community Association (OCA)",
+    "category": "Product",
+    "website": "https://github.com/OCA/product-attribute",
+    "depends": ["product"],
+    "data": [
```
Comment (nikul-serpentcs) [inline product_code_unique/__manifest__.py]: @bizzappdev if no need then remove it

File: product_code_unique/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    ],
+    'demo': [
```
Comment (nikul-serpentcs) [inline product_code_unique/__manifest__.py]: same here

**Review by nikul-serpentcs — COMMENTED** [review]
Improve Code

**Comment by bizzappdev** [comment]
changes, as suggested, are done.

**Comment by bizzappdev** [comment]
@nikul-serpentcs Please review again.

**Comment by nikul-serpentcs** [comment]
> @nikul-serpentcs Please review again.

@bizzappdev Squash commits

**Review by alan196 — APPROVED** [review]
Code review 👍 just squash the commits

**Comment by bizzappdev** [comment]
squash Of the commits is done.

**Review by nikul-serpentcs — APPROVED** [review]
Code Review LGTM :+1: 

README like  https://github.com/OCA/maintainer-tools/tree/master/template/module/readme

**Pull request merged**